polkadot/node/subsystem-bench/src/availability/mod.rs

```diff
@@ -0,0 +1,387 @@
+
+				candidate_receipt
+			})
+			.collect::<Vec<_>>()
+			.into_iter()
+			.cycle();
+	}
+
+	pub fn new(config: &TestConfiguration) -> Self {
+		let config = config.clone();
+
+		let mut chunks = Vec::new();
+		let mut available_data = Vec::new();
+		let mut candidate_receipt_templates = Vec::new();
+		let mut pov_size_to_candidate = HashMap::new();
+
+		// we use it for all candidates.
+		let persisted_validation_data = PersistedValidationData {
+			parent_head: HeadData(vec![7, 8, 9]),
+			relay_parent_number: Default::default(),
+			max_pov_size: 1024,
+			relay_parent_storage_root: Default::default(),
+		};
+
+		// For each unique pov we create a candidate receipt.
+		for (index, pov_size) in config.pov_sizes().iter().cloned().unique().enumerate() {
+			gum::info!(target: LOG_TARGET, index, pov_size, "{}", "Generating template candidate".bright_blue());
+
+			let mut candidate_receipt = dummy_candidate_receipt(dummy_hash());
+			let pov = PoV { block_data: BlockData(vec![index as u8; pov_size]) };
+
+			let new_available_data = AvailableData {
+				validation_data: persisted_validation_data.clone(),
+				pov: Arc::new(pov),
+			};
+
+			let (new_chunks, erasure_root) = derive_erasure_chunks_with_proofs_and_root(
+				config.n_validators,
+				&new_available_data,
+				|_, _| {},
+			);
+
+			candidate_receipt.descriptor.erasure_root = erasure_root;
+
+			chunks.push(new_chunks);
+			available_data.push(new_available_data);
+			pov_size_to_candidate.insert(pov_size, index);
+			candidate_receipt_templates.push(candidate_receipt);
+		}
+
+		let pov_sizes = config.pov_sizes().to_vec().into_iter().cycle();
+		gum::info!(target: LOG_TARGET, "{}","Created test environment.".bright_blue());
+
+		let mut _self = Self {
+			config,
+			available_data,
+			candidate_receipt_templates,
+			chunks,
+			pov_size_to_candidate,
+			pov_sizes,
+			candidate_hashes: HashMap::new(),
+			candidates: Vec::new().into_iter().cycle(),
+		};
+
+		_self.generate_candidates();
+		_self
+	}
+}
+
+fn derive_erasure_chunks_with_proofs_and_root(
+	n_validators: usize,
+	available_data: &AvailableData,
+	alter_chunk: impl Fn(usize, &mut Vec<u8>),
+) -> (Vec<ErasureChunk>, Hash) {
+	let mut chunks: Vec<Vec<u8>> = obtain_chunks(n_validators, available_data).unwrap();
+
+	for (i, chunk) in chunks.iter_mut().enumerate() {
+		alter_chunk(i, chunk)
+	}
+
+	// create proofs for each erasure chunk
+	let branches = branches(chunks.as_ref());
+
+	let root = branches.root();
+	let erasure_chunks = branches
+		.enumerate()
+		.map(|(index, (proof, chunk))| ErasureChunk {
+			chunk: chunk.to_vec(),
+			index: ValidatorIndex(index as _),
+			proof: Proof::try_from(proof).unwrap(),
+		})
+		.collect::<Vec<ErasureChunk>>();
+
+	(erasure_chunks, root)
+}
+
+pub async fn benchmark_availability_read(env: &mut TestEnvironment, mut state: TestState) {
+	let config = env.config().clone();
+
+	env.import_block(new_block_import_info(Hash::repeat_byte(1), 1)).await;
+
```

Comment:
This is not part of launching a test, calling this will make the overseer send a `ActiveLeavesUpdate` signal to all subsystems. Also the block import information might be generated as well to make sense with gossip messages like in the case of approval voting, where on each assignment we have the block hash and candidate index